### DISCUSSION TITLE
Синтаксис HTTPS и SVCB записей в dnsrewrite

### DISCUSSION THREAD

**Discussion by ammnt** [AdguardTeam/AdGuardHome#3410] [Q&A] [answered]
Привет,

помогите, пожалуйста, разобраться с синтаксисом HTTPS и SVCB записей через dnsrewrite:

```
||dns1.ammnt.app^$dnsrewrite=NOERROR;HTTPS;1 dns2.ammnt.app alpn=h2 ipv4hint=82.146.52.66 ipv6hint=2a01:231:4:fc::2
||dns2.ammnt.app^$dnsrewrite=NOERROR;SVCB;1 dns1.ammnt.app alpn=h2 ipv4hint=82.146.12.63 ipv6hint=2a02:232:4:fc::2
```
(это просто для тесты - адреса и домены не настоящие)

Это верно составленное правило? Можно ли использовать модификатор important для них? Я спрашиваю, потому что в логе запись возвращает только NOERROR - самого значения нет и проверить нельзя.

Спасибо.

**Comment by ameshkov** [comment]
@ammnt так вам надо проверять не в логе, а утилитой типа dig или dnslookup делать запрос типа HTTPS и смотреть вернуло ли что-нибудь.

`RRTYPE=https dnslookup dns1.ammnt.app x.x.x.x`

где `x.x.x.x` - адрес вашего AG Home.

**Reply by ammnt** [reply]
@ameshkov, спасибо за ответ. Тут вопрос в большей степени про синтаксис😀Он верный? Модификатор important можно использовать?

**Reply by ameshkov** [reply]
Да вот сложно сказать, ну вроде бы верный:)

**Reply by ameshkov** [reply]
$important можно использовать

**Reply by ammnt** [reply]
@ameshkov, спасибо! А по умолчанию из коробки сервер HTTPS и SVCB не возвращает никакие для DoH-протокола?

**Reply by ammnt** [reply]
Короче я так и не смог заставить DNS-сервер возвращать HTTPS-запись для себя самого - для своего же хоста. Ни через dnsrewrite, ни с IPv6 ни без, ни с модификатором ни без. Никак...

**Reply by ammnt** [reply]
Проверял я, разумеется, с внешней стороны - kdig'ом и вашей тулзой 😀

**Comment by ainar-g** [comment, marked as answer]
У меня всё работает и синтаксис верный.  Вот программа на го, которая успешно резолвит _HTTPS_-запись: https://play.golang.org/p/gXnuLFqFV-Z.

~~`dig` и `nslookup` у вас вряд ли заработают, так как этот тип очень новый.~~

Поправка:

```sh
dig -t TYPE65 'blog.cloudflare.com' @localhost
```

```none
; <<>> DiG 9.16.1-Ubuntu <<>> -t TYPE65 blog.cloudflare.com @localhost
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 13337
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags: do; udp: 4096
;; QUESTION SECTION:
;blog.cloudflare.com.           IN      TYPE65

;; ANSWER SECTION:
blog.cloudflare.com.    110     IN      TYPE65  \# 79 000100000100180268330568332D32390568332D32380568332D3237 0268320004000868121A2E68121B2E00060020260647000000000000 00000068121A2E26064700000000000000000068121B2E

;; Query time: 20 msec
;; SERVER: 127.0.0.1#54(127.0.0.1)
;; WHEN: Wed Sep 01 20:40:28 MSK 2021
;; MSG SIZE  rcvd: 139
```
